### PR TITLE
Grafana: Upgrades mysql images from 5.7 to 8

### PR DESCRIPTION
**Pull request by owensmallwood**
Adding a build step for mysql8 since we're transitioning to that in cloud.

[Enterprise PR](https://github.com/grafana/grafana-enterprise/pull/5016)

### CODE REVIEW

**Review by papagian — APPROVED** [review]
There some additional references of the `5.7.39` version that we need to address here or in a follow-up:
- the default version in `mysql` devenv [block](https://github.com/grafana/grafana/blob/3bfb50f989d11eb9cb5c3da8eaaf5bf7aa1c90b5/devenv/docker/blocks/mysql/.env#L1)
- the default version in `mysql_tests` devenv [block](https://github.com/grafana/grafana/blob/3bfb50f989d11eb9cb5c3da8eaaf5bf7aa1c90b5/devenv/docker/blocks/mysql_tests/.env#L1)
- the default argument for the `mysql_tests` [Dockerfile](https://github.com/grafana/grafana/blob/3bfb50f989d11eb9cb5c3da8eaaf5bf7aa1c90b5/devenv/docker/blocks/mysql_tests/Dockerfile#L1)

**Comment by sakjur** [comment]
🤔 I'm a bit worried that we could end up depending on MySQL 8 features in a migration somewhere without detecting it before shipping since we've committed to supporting MySQL 5.7 (https://grafana.com/docs/grafana/v9.5/setup-grafana/installation/#supported-databases) until it's no longer supported by upstream.

File: .drone.yml
Comment (sakjur) [inline .drone.yml:1]: This file needs to be generated with `make drone`.

**Comment by papagian** [comment]
> 🤔 I'm a bit worried that we could end up depending on MySQL 8 features in a migration somewhere without detecting it before shipping since we've committed to supporting MySQL 5.7 (https://grafana.com/docs/grafana/v9.5/setup-grafana/installation/#supported-databases) until it's no longer supported by upstream.

that's also my concern: ideally we would like to run integration tests for against `5.7` and `8.0`

**Comment by kminehart** [comment]
> ideally we would like to run integration tests both against 5.7 and 8.0

Good idea!

**Comment by owensmallwood** [comment]
@sakjur @kminehart @papagian  How much work would it be to have our mysql integration tests run against 5.7 and 8? I'm not really familiar with our build pipepine. This PR was mostly to see what would break if we bump the mysql image to 8.

**Comment by sakjur** [comment]
I don't think that would be a lot of work, I'll create a branch to try it out quickly

**Comment by sakjur** [comment]
Got it working in #67710.

**Comment by owensmallwood** [comment]
Thanks @sakjur . I'll do the same thing to the [enterprise PR](https://github.com/grafana/grafana-enterprise/pull/5016) for this

**Comment by owensmallwood** [comment]
@sakjur Do you know how long we're committed to supporting 5.7? Assuming until EOL in October?

**Comment by sakjur** [comment]
Yup, October.

**Pull request merged**